js-api-spec/value/list.test.ts

```diff
@@ -0,0 +1,283 @@
+        const list = new SassList([
+          new SassString('a'),
+          new SassString('b'),
+          new SassString('c'),
+        ]);
+        expect(list.separator).toBe(',');
+        expect(list.hasBrackets).toBe(false);
+      });
+
+      it('supports null separators', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {separator: null}
+        );
+        expect(list.separator).toBe(null);
+      });
+
+      it('supports space separators', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {separator: ' '}
+        );
+        expect(list.separator).toBe(' ');
+      });
+
+      it('supports slash separators', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {separator: '/'}
+        );
+        expect(list.separator).toBe('/');
+      });
+
+      it('supports brackets', () => {
+        const list = new SassList(
+          [new SassString('a'), new SassString('b'), new SassString('c')],
+          {brackets: true}
+        );
+        expect(list.hasBrackets).toBe(true);
+      });
+    });
+
+    describe('single-element list', () => {
+      const list = new SassList([new SassNumber(1)]);
+
+      it('has a comma separator', () => {
+        expect(list.separator).toBe(',');
+      });
+
+      it('has no brackets', () => {
+        expect(list.hasBrackets).toBe(false);
+      });
+
+      it('returns its contents as a list', () => {
+        expect(list.asList.equals(List([new SassNumber(1)]))).toBe(true);
+      });
+    });
+
+    describe('a scalar value', () => {
+      const string = new SassString('blue');
+
+      it('has a null separator', () => {
+        expect(string.separator).toBe(null);
+      });
+
+      it('has no brackets', () => {
+        expect(string.hasBrackets).toBe(false);
+      });
+
+      it('returns itself as a list', () => {
+        const list = string.asList;
+        expect(list.size).toBe(1);
+        expect(list.get(0)).toBe(string);
+      });
+
+      describe('Sass to JS index conversion', () => {
+        it('converts a positive index', () => {
+          expect(string.sassIndexToListIndex(new SassNumber(1))).toBe(0);
+        });
+
+        it('converts a negative index', () => {
+          expect(string.sassIndexToListIndex(new SassNumber(-1))).toBe(0);
+        });
+
+        it('rejects invalid indices', () => {
+          expect(() =>
+            string.sassIndexToListIndex(new SassNumber(0))
+          ).toThrow();
+          expect(() =>
+            string.sassIndexToListIndex(new SassNumber(2))
+          ).toThrow();
+          expect(() =>
+            string.sassIndexToListIndex(new SassNumber(-2))
+          ).toThrow();
+        });
+      });
+    });
+
+    describe('an empty list', () => {
```

Comment:
Test that this equals another call to `SassList.empty()` and a call to `new SassList()` without any elements.